### PR TITLE
Feature/#98 feedbackonbuttons

### LINKED ISSUE #98: Optional: It would be nice to have some feedback upon clicking the buttons within the providers test app. Such as, if you click there button, it should output some text that says 'Message sent' or 'Error sending message, check API key is valid'
It would be nice to have some feedback upon clicking the buttons within the providers test app. Such as, if you click there button, it should output some text that says 'Message sent' or 'Error sending message, check API key is valid'

_Originally posted by @sumitramanga in https://github.com/MindscapeHQ/raygun4android/issues/95#issuecomment-1809402050_

### PR DESCRIPTION
**Pull request by TheRealAgentK**
### Description :memo:

Fixes #98 

Adds some Snackbar notification on actions in the sample app.

**Type of change**

*Delete options that are not relevant.*

- [x] New feature (non-breaking change which adds functionality)

**Updates**

:point_right: Added Material UI lib
:point_right: Added Snackbars on button actions, resource strings in `strings.xml`

### Author to check :eyeglasses:
- [x] Project and all contained modules build
- [x] Tested on appropriate devices/emulators and SDK versions
~- [] Reviewed by another developer~
N/A
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Change has been tested on a device/emulator
- [ ] Code is written to standards
~- [ ] Appropriate tests have been written (code comments, internal docs)~
N/A

### CODE REVIEW

File: app/src/main/res/layout/activity_second.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".MainActivity">
+    tools:context=".SecondActivity">
```
Comment (sumitramanga) [inline app/src/main/res/layout/activity_second.xml:7]: Good catch 👍🏽

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -11,4 +11,10 @@
     <string name="go_to_a_second_activity">Go to a second Activity</string>
     <string name="hello_world_main_activity">Hello World! Main Activity!</string>
     <string name="hello_world_second_activity">Hello World! Second Activity!</string>
+    <string name="you_have_just_sent_an_error_with_raygun4android">You have just sent an error with Raygun4Android.</string>
+    <string name="you_just_created_and_caught_an_exception">You just created and caught an exception.</string>
+    <string name="user_is_now_set_to_anonymous_for_future_raygun_reports">User is now set to \'Anonymous\' for future Raygun reports.</string>
+    <string name="user_is_now_set_to_user_a_for_future_raygun_reports">User is now set to \'User A\' for future Raygun reports.</string>
+    <string name="user_is_now_set_to_user_b_for_future_raygun_reports">User is now set to \'User B\' for future Raygun reports.</string>
+    <string name="we_re_now_on_the_second_activity_screen">We\'re now on the second activity screen.</string>
```
Comment (sumitramanga) [inline app/src/main/res/values/strings.xml:19]: Nitpick: some long string names, is that usual in this space? wondering if it's ok to shorten them or does it not make sense?

File: app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt [hunk 3]
```diff
@@ -3,15 +3,16 @@ package com.raygun.raygun4android.sample
...
 import android.widget.Button
 import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
 import com.raygun.raygun4android.RaygunClient
 import com.raygun.raygun4android.messages.crashreporting.RaygunBreadcrumbLevel
 import com.raygun.raygun4android.messages.crashreporting.RaygunBreadcrumbMessage
 import com.raygun.raygun4android.messages.shared.RaygunUserInfo
-import java.util.*
+import java.util.WeakHashMap
```
Comment (sumitramanga) [inline app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt:17]: Where is this used?

File: app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt [hunk 4]
```diff
@@ -3,15 +3,16 @@ package com.raygun.raygun4android.sample
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
-import androidx.appcompat.app.AppCompatActivity
 import android.util.Log
 import android.widget.Button
 import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
```
Comment (sumitramanga) [inline app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt:12]: Love it 💯 Do you have screenshots/gif/screen recording of how it looks?

**Review by sumitramanga — CHANGES_REQUESTED** [review]
Just some comments to address but otherwise mostly looks good! Thank you for implementing this change ⭐

**Comment by sumitramanga** [comment]
Hi @TheRealAgentK ! Just wanted to check in to see if you'll have time to work on this? :)

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (TheRealAgentK) [reply app/src/main/res/values/strings.xml:19]: Yeah, that is quite common in general.

File: app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt (hunk 3 above)
Comment (TheRealAgentK) [reply app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt:17]:
```
val customData = WeakHashMap<String,Any>()
customData["someKey"] = "someValue"
customData["someotherkey"] = "someothervalue"
```

File: app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt (hunk 4 above)
Comment (TheRealAgentK) [reply app/src/main/java/com/raygun/raygun4android/sample/SecondActivity.kt:12]: Will create some and put into the repo

**Review by miquelbeltran — APPROVED** [review]

File: app/src/main/java/com/raygun/raygun4android/sample/MainActivity.kt [hunk 5]
```diff
@@ -60,40 +64,52 @@ class MainActivity : AppCompatActivity() {
...
                 val i = 4
                 Log.d("Raygun4Android-Sample", "This is here purely so that our alternative value for i gets used and not optimised away in a release build: $i")
+                Snackbar.make(it, getString(R.string.you_just_created_and_caught_an_exception),Snackbar.LENGTH_SHORT).show()
             }
         }
 
         buttonSetUserAnon.setOnClickListener {
             val user = RaygunUserInfo()
             RaygunClient.setUser(user)
+            Snackbar.make(it, getString(R.string.user_is_now_set_to_anonymous_for_future_raygun_reports),Snackbar.LENGTH_SHORT).show()
```
Comment (miquelbeltran) [inline app/src/main/java/com/raygun/raygun4android/sample/MainActivity.kt:74]:
```suggestion
            Snackbar.make(it, getString(R.string.user_is_now_set_to_anonymous_for_future_raygun_reports), Snackbar.LENGTH_SHORT).show()
```

Small thing, just a formatting change. It's also present in the `Snackbar.make` calls.

Comment (TheRealAgentK) [reply app/src/main/java/com/raygun/raygun4android/sample/MainActivity.kt:74]: I might leave it as is for now, one of the next things this needs is a linter setup and formatting at build etc anyway.

**Pull request merged**